### PR TITLE
fix: only select with main mouse button

### PR DESCRIPTION
**Pull request by T-Wizard**
(maybe also abort selection if multiple buttons are pressed?)

## Motivation

fix qlik-oss/sn-list-objects#68

## Requirements checklist

- [ ] Api specification
  - [ ] Ran `yarn spec`
    - [ ] No changes **_OR_** API changes has been formally approved
- [ ] Unit/Component test coverage
- [ ] Correct PR title for the changes (fix, chore, feat)

When build and tests have passed:

- [ ] Add code reviewers, for example @qlik-oss/nebula-core

### CODE REVIEW

File: apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx [hunk 1]
```diff
@@ -379,10 +379,6 @@ function RowColumn({ index, rowIndex, columnIndex, style, data }) {
   );
 
   const preventContextMenu = (event) => {
-    if (checkboxes) {
-      // Event will not propagate in the checkbox/radiobutton case
-      onClick(event);
-    }
```
Comment (T-Wizard) [inline apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx:385]: @johanlahti do you remember why this call to onClick was added? (in #808)

**Comment by T-Wizard** [comment]
Anyone that have any ide why the unit test fails?
I do work when I run it locally on my machine.

**Review by DanielS-Qlik — APPROVED** [review]

**Comment by Caele** [comment]
@T-Wizard I've made changes to this in my react18 update as well. The main problem there was the usage of onClick for the checkboxes. Basically a label/input pair will fire double click events when you click one of them, which with the react update somehow manifested. I've changed it to use onChange instead which seems to work better.
Maybe this could wait until that update?

File: apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx (hunk 1 above)
Comment (johanlahti) [reply apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx:385]: Maybe it could be implemented differently instead of this workaround.

Comment (Caele) [reply apis/nucleus/src/components/listbox/components/ListBoxRowColumn.jsx:385]: Looking into this now as well, got a bit messy with the React update

**Review by Caele — APPROVED** [review]

**Pull request merged**